src/main/java/org/folio/circulation/domain/UpdateItem.java

```diff
@@ -28,6 +29,13 @@ public UpdateItem(Clients clients) {
 
     return of(() -> changeItemOnCheckIn(item, requestQueue, checkInServicePointId))
       .after(updatedItem -> {
+        if (updatedItem == null) {
```

Comment:
Can we do the validation before it gets here? Somewhere in CreateRequestService.refuseWhenItemIsNotValid(), for example? If we do it there then logic may have to be shared among requests, loans, renewals, etc...

Reply:
I think this is the earliest place in the check in process where we have the necessary context and have retrieved all the supporting objects to determine that the highest fulfillable request is missing the pickup service point.

Admittedly, I don't like using `null` here to convey that case. But I wanted to introduce minimal change due to the fact that this really shouldn't happen. This US was to protect the check in call from bad data, however it gets in there.

I believe we do need to add validation into the `CreateRequestService` that prevents a request with fulfillment to the hold shelf from being created without a pickup service point. I'm not sure this is the right US to do that or if this is agreed upon. There is a UI component to the request creation change or at least there should be.

I'm not sure about the shared logic, so I might be missing something. Maybe you can elaborate?

Reply:
That's fine if it's the earliest place. I thought it could be done earlier in the process.  My thought about shared logic is if we implemented at some place earlier, other workflows may not benefit from it so we'd have to make this check reusable/shared among other workflows.  I called out CreateRequestService as an example because that's the other place that I thought we could implement it, and which I mistakenly thought that this JIRA issue is also addressing.